### PR TITLE
Adding g2o-opencv python package

### PR DESCRIPTION
**Pull request by miquelmassot**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from an official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Hi conda team!

I'm trying to set up my first release to conda-forge for a g2o python bindings wrapper. I'd kindly request some guidance to follow up on this if I get stuck.

@conda-forge/help-python-c

### CODE REVIEW

File: recipes/g2o-python/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (dopplershift) [inline recipes/g2o-python/LICENSE:1]: The license is already included in the source tarball so you don't need to (and shouldn't) include it in the recipe. Please go ahead and remove.

File: recipes/g2o-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
+    - numpy
+    - eigen
+    - scikit-build
+  run:
+    - python
+    - numpy
+    - pip
```
Comment (dopplershift) [inline recipes/g2o-python/meta.yaml:40]: I don't think pip is required at runtime:
```suggestion
```

File: recipes/g2o-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  build:
+    - "{{ compiler('cxx') }}"
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
+    - python
+    - numpy
+    - pip
+    - eigen
+    - scikit-build
```
Comment (dopplershift) [inline recipes/g2o-python/meta.yaml:26]: I think these are only needed on host:
```suggestion
```

File: recipes/g2o-python/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  host:
+    - python
+    - pip
+    - "{{ compiler('cxx') }}"
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
```
Comment (dopplershift) [inline recipes/g2o-python/meta.yaml:33]: Conversely, only on build:
```suggestion
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Some minor adjustments to the recipe.

**Comment by dopplershift** [comment]
Ok, it's not clear to my why the build is failing. @conda-forge/help-c-cpp ?

**Comment by miquelmassot** [comment]
@conda-forge/help-c-cpp

**Comment by dopplershift** [comment]
Here's the build failure for anyone curious:
```
      skbuild.exceptions.SKBuildInvalidFileInstallationError:
        CMake-installed files must be within the project root.
          Project Root  : /home/conda/staged-recipes/build_artifacts/g2o-python_1675853201793/work/_skbuild/linux-x86_64-3.10/cmake-install
          Violating File: /home/conda/staged-recipes/build_artifacts/g2o-python_1675853201793/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/include/g2o/config.h
```

File: recipes/g2o-python/LICENSE (hunk 1 above)
Comment (carterbox) [reply recipes/g2o-python/LICENSE:1]: This file is still here. Why resolve this comment?

**Review by carterbox — CHANGES_REQUESTED** [review]
This project is revending g2o which, in turn, is revending ceres and freeglut. You need to collect all of the licenses from these subprojects into the license section of this recipe. This license requirement is mentioned in the checklist!

**Comment by carterbox** [comment]
> Here's the build failure for anyone curious:
> 
> ```
>       skbuild.exceptions.SKBuildInvalidFileInstallationError:
>         CMake-installed files must be within the project root.
>           Project Root  : /home/conda/staged-recipes/build_artifacts/g2o-python_1675853201793/work/_skbuild/linux-x86_64-3.10/cmake-install
>           Violating File: /home/conda/staged-recipes/build_artifacts/g2o-python_1675853201793/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/include/g2o/config.h
> ```

This build failure is from scikit-build. It says that the project is trying to install files outside of python's site-packages directory. This is not allowed because python packages shouldn't be installing files into $PREFIX/include. Python packages don't ship headers.

**Pull request closed without merging**